lib/side_header_list_view.dart

```diff
@@ -21,97 +19,90 @@ class SideHeaderListView extends StatefulWidget {
+    @required this.itemExtent,
     @required this.headerBuilder,
     @required this.itemBuilder,
     @required this.hasSameHeader,
     this.padding,
-  })
-      : super(key: key);
+  }) : super(key: key);
 
   @override
-  _SideHeaderListViewState createState() => new _SideHeaderListViewState();
+  _SideHeaderListViewState createState() => _SideHeaderListViewState();
 }
 
 class _SideHeaderListViewState extends State<SideHeaderListView> {
   int currentPosition = 0;
+  final _controller = ScrollController();
 
   @override
-  Widget build(BuildContext context) {
-    return new Stack(
-      children: <Widget>[
-        new Positioned(
-          child: new Opacity(
-            opacity: _shouldShowHeader(currentPosition) ? 0.0 : 1.0,
-            child: widget.headerBuilder(context, currentPosition >= 0 ? currentPosition : 0),
-          ),
-          top: 0.0 + (widget.padding?.top ?? 0),
-          left: 0.0 + (widget.padding?.left ?? 0),
-        ),
-        new ListView.builder(
-            padding: widget.padding,
-            itemCount: widget.itemCount,
-            itemExtent: widget.itemExtend,
-            controller: _getScrollController(),
-            itemBuilder: (BuildContext context, int index) {
-              return new Row(
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: <Widget>[
-                  new FittedBox(
-                    child: new Opacity(
-                      opacity: _shouldShowHeader(index) ? 1.0 : 0.0,
-                      child: widget.headerBuilder(context, index),
-                    ),
-                  ),
-                  new Expanded(child: widget.itemBuilder(context, index))
-                ],
-              );
-            }),
-      ],
-    );
+  void initState() {
+    super.initState();
+    // Reposition side view each time the main list view is scrolled
+    _controller.addListener(_reposition);
+    // In case the initial offset is not 0, we reposition the side view to the
+    // correct offset after first build
+    // TODO: this produces a flickering effect -- we should try to position it
+    // correctly before the first build
+    WidgetsBinding.instance.addPostFrameCallback((_) => _reposition());
   }
 
-  bool _shouldShowHeader(int position) {
-    if(position < 0){
-      return true;
-    }
-    if (position == 0 && currentPosition < 0) {
-      return true;
-    }
-
-    if (
-      position != 0 &&
-      position != currentPosition &&
-        !widget.hasSameHeader(position, position - 1)) {
-      return true;
-    }
+  @override
+  void dispose() {
+    _controller?.removeListener(_reposition);
+    super.dispose();
+  }
 
-    if (
-      position != widget.itemCount -1 &&
-      !widget.hasSameHeader(position, position + 1) &&
-        position == currentPosition) {
-      return true;
-    }
-    return false;
+  void _reposition() {
+    print('new position = ${(_controller.offset / widget.itemExtent).floor()}');
```

Comment:
I don't think this print is a good idea.
```suggestion
```

Reply:
yep that was an old commit (I push another one since then)